haystack/preprocessor/utils.py

```diff
@@ -86,32 +90,46 @@ def convert_files_to_dicts(dir_path: str, clean_func: Optional[Callable] = None,
     """
 
     file_paths = [p for p in Path(dir_path).glob("**/*")]
-    if ".pdf" in [p.suffix.lower() for p in file_paths]:
-        pdf_converter = PDFToTextConverter()  # type: Optional[PDFToTextConverter]
-    else:
-        pdf_converter = None
+    allowed_suffixes = [".pdf", ".txt", ".docx"]
+    converter_dict: Dict[str, BaseConverter] = {}
 
-    documents = []
+    file_ext_dict: Dict[str, List[Path]] = {}
```

Comment:
I'd sugget a few renaming to make the code more readable:
 `file_ext_dict` => `suffix2paths` 
`converter_dict` => `suffix2converter`

Reply:
done